packages/shared/src/components/post/PostEngagements.tsx

```diff
@@ -58,10 +65,23 @@ function PostEngagements({
     closeShareComment,
     onShowShareNewComment,
   } = useShareComment(analyticsOrigin, enableShowShareNewComment);
+  const [isJoinSquadBannerDismissed] = usePersistentContext(
+    SQUAD_COMMENT_JOIN_BANNER_KEY,
+    false,
+  );
```

Comment:
I used this kind of thing for lego campaign and it worked fine, not sure about the edge case.

Reply:
Not sure if it was Lego, but I refactored some use-case like this when you where on leave.
It's pretty much if two spots overlap and try to open the same context, one will reset to default. (or at least that's the experience)

Reply:
Ok, let's change it then, what would the fix be here, how did you refactor other use cases?

Reply:
Basically making sure there is only one origin that can call the same `usePersistentContext`.
So in this case it would mean creating a wrapping hook for this one.

`useJoinSquadBannerDismissed` that would simply wrap for both places, that should do the trick I think.

Reply:
So if I understand correctly you are saying we need to mount only one instance of the hook to not trigger supposed edge case?

Reply:
Yes correct, I didn't really deep dive into it but as long as it's only 1 instance it doesn't happen.

Reply:
Ok, then the hook would not be a solution since again calling `useJoinSquadBannerDismissed` multiple times would basically mount `usePersistentContext` multiple times again. It would have to be prop drill of the value.

Reply:
I don't want this task to drag on too long so we can maybe launch as is and see if it causes these?
From what I recall was init it in a hook, which would use it more than once was fine. It was the state duplication which was the issue.

but let's rather see if it occurs?

Reply:
Yes, let's jump on it then 🚀 🙏